plugins/gimp/file-jxl-save.cc

```diff
@@ -643,6 +675,52 @@ bool JpegXlSaveOpts::SetPrecision(int gimp_precision) {
   return true;
 }  // JpegXlSaveOpts::SetPrecision
 
+class GimpParser {
+ public:
+  JxlFrameHeader LayerNameToFrameHeader(gchar* layer_name) {
+    JxlFrameHeader frame_header;
+    JxlEncoderInitFrameHeader(&frame_header);
+    gchar** remainder = &layer_name;
+    LayerNameToJxlLayerName(layer_name, remainder);
+    if (g_str_has_prefix(*remainder, " (")) {
+      gchar* remainder2;
+      gchar** endptr = &remainder2;
+      gint64 ms_len = g_ascii_strtoll(*remainder + 2, endptr, 10);
+      if (ms_len == 0) {
+        ms_len = 1;
+      }
+      frame_header.duration = ms_len;
+    }
+    gchar* found;
+    gsize len = g_ref_string_length(*remainder);
+    found = g_strstr_len(*remainder, len, GIMP_BLEND_REPLACE_FLAG->str);
+    if (found != nullptr) {
+      frame_header.layer_info.blend_info.blendmode = JXL_BLEND_REPLACE;
+    }
+    found = g_strstr_len(*remainder, len, GIMP_BLEND_COMBINE_FLAG->str);
+    if (found != nullptr) {
+      frame_header.layer_info.blend_info.blendmode = JXL_BLEND_ADD;
```

Comment:
I think this should be JXL_BLEND_BLEND. That's the normal alpha blending that should correspond to "combine".

JXL_BLEND_ADD numerically adds the RGBA sample values (including alpha values), which is not what you want to do here.

Reply:
thank you, i'm sure you're right